cli/PrintToScreen.cpp

```diff
@@ -147,4 +148,25 @@ void PrintToScreen::printTuple(const TupleStorageSubBlock &tuple_store,
   fputc('\n', out);
 }
 
+std::size_t PrintToScreen::GetNumTuplesInRelation(
+    const CatalogRelation &relation, StorageManager *storage_manager) {
+  const std::vector<block_id> &blocks = relation.getBlocksSnapshot();
+  std::size_t total_num_tuples = 0;
+  for (block_id block : blocks) {
+    total_num_tuples +=
+        storage_manager->getBlock(block, relation)->getNumTuples();
+  }
+  return total_num_tuples;
```

Comment:
@zuyu I had considered implementing printing the output size inside the ``PrintRelation()`` but didn't go ahead with the approach for following reasons: 

- The number of rows is not the part of the relation per se. 
- A lot of tests in optimizer and some in cli modules depend on the PrintRelation for correctness. Modifying all of them is a fair bit of work. 
- As @pateljm mentioned, separating these functions help to control whether we want to print both the output and its size, or just the size of the output. 
- This particular function can be reusable, e.g. for the describe table and describe database commands implemented by @rogersjeffreyl , we can print the number of tuples in the tables as a part of the output. 
-  The eviction issue you had mentioned is valid, but for the common case (e.g. TPC-H, SSB workloads) the query output size is typically small i.e. few number of blocks. So eviction won't be an issue in the common case. A permanent fix will be to cache the number of tuples for each relation, but that's for the future.